src/components/ui/AddressCombobox.tsx

```diff
@@ -0,0 +1,126 @@
+import {
+  Combobox,
+  ComboboxInput,
+  ComboboxOption,
+  ComboboxOptions,
+} from '@headlessui/react'
+import { Dispatch, SetStateAction, useEffect, useState } from 'react'
+import { useConfig } from '@/hooks/useConfig'
+import { getSuggestedAddresses } from '@/services/location/address'
+
+// Import the Select Combobox component for the side-effects of injecting CSS
+// for related components, such as Textbox and Listbox.
+import '@utrecht/select-combobox-react/dist/css'
+import { useFormStore } from '@/store/form_store'
+import { Address } from '@/types/form'
+
+type AddressComboboxProps = {
+  updatePosition?: (lat: number, lng: number, flyTo?: boolean) => void
+  setIsMapSelected?: Dispatch<SetStateAction<boolean>>
+}
+
+export const AddressCombobox = ({
+  updatePosition,
+  setIsMapSelected,
+}: AddressComboboxProps) => {
+  const [query, setQuery] = useState('')
+  const { config } = useConfig()
+  const [addressOptions, setAddressOptions] = useState<any[]>([])
+  const { formState, updateForm } = useFormStore()
+
+  const parsePoint = (str: string): [number, number] | undefined => {
+    const match = /POINT\((\d+(?:\.\d+)?)\s+(\d+(?:\.\d+)?)\)/i.exec(str)
+
+    if (match) {
+      return [parseFloat(match[1]), parseFloat(match[2])]
+    }
+  }
+
+  const normalizeQuery = (str: string) => str.trim().replace(/\s+/, ' ')
+
+  useEffect(() => {
+    const municipality = config ? config.base.municipality : ''
+    const normalizedQuery = normalizeQuery(query)
+
+    const getAddressOptions = async () => {
+      if (query.length >= 1) {
+        const apiCall = await getSuggestedAddresses(
+          normalizedQuery,
+          municipality
+        )
+
+        const options = apiCall.response.docs.map((item) => ({
+          coordinates: parsePoint(item.centroide_ll),
+          id: item.id,
+          postcode: item.postcode,
+          huisnummer: item.huis_nlt,
+          woonplaats: item.woonplaatsnaam,
+          openbare_ruimte: item.straatnaam,
+          weergave_naam: item.weergavenaam,
+        }))
+
+        setAddressOptions(options)
+
+        return
+      }
+
+      setAddressOptions([])
+    }
+
+    getAddressOptions()
+  }, [query])
+
+  const onChangeAddress = (selectedAddress: Address) => {
+    if (selectedAddress && updatePosition) {
+      updatePosition(
+        selectedAddress.coordinates[1],
+        selectedAddress.coordinates[0]
+      )
+    }
+
+    updateForm({
+      ...formState,
+      address: selectedAddress,
+      coordinates: [
+        selectedAddress.coordinates[1],
+        selectedAddress.coordinates[0],
+      ],
+    })
+
+    if (setIsMapSelected) {
+      setIsMapSelected(true)
+    }
+  }
+
+  return (
+    <Combobox
+      value={formState.address}
+      onChange={onChangeAddress}
+      onClose={() => setQuery('')}
+    >
+      <ComboboxInput
```

Comment:
als er een adres is gekozen of ingevuld en je maakt het adres leeg krijg je een foutmelding:

<img width="638" alt="image" src="https://github.com/user-attachments/assets/b9d8c367-888b-48c9-96aa-8eb2129c3f39">


```
AddressCombobox.tsx:85 Uncaught TypeError: Cannot read properties of null (reading 'coordinates')
    at onChangeAddress (AddressCombobox.tsx:85:25)
    at Object.eval [as current] (use-controllable.js:13:103)
    at eval (use-event.js:17:18)
    at 0 (combobox.js:496:45)
    at u (match.js:11:41)
    at Object.eval [as current] (combobox.js:494:140)
    at Object.eval [as onChange] (use-event.js:17:18)
    at Object.eval [as current] (combobox.js:579:11)
    at eval (use-event.js:17:18)
    at Object.eval [as current] (combobox.js:659:69)
    at eval (use-event.js:17:18)
    at onChange (render.js:175:42)
    at onChange (render.js:153:17)
    at HTMLUnknownElement.callCallback (react-dom.development.js:20565:14)
    at Object.invokeGuardedCallbackImpl (react-dom.development.js:20614:16)
    at invokeGuardedCallback (react-dom.development.js:20689:29)
    at invokeGuardedCallbackAndCatchFirstError (react-dom.development.js:20703:25)
    at executeDispatch (react-dom.development.js:32128:3)
    at processDispatchQueueItemsInOrder (react-dom.development.js:32160:7)
    at processDispatchQueue (react-dom.development.js:32173:5)
    at dispatchEventsForPlugins (react-dom.development.js:32184:3)
    at eval (react-dom.development.js:32374:12)
    at batchedUpdates$1 (react-dom.development.js:24953:12)
    at batchedUpdates (react-dom.development.js:28844:12)
    at dispatchEventForPluginEventSystem (react-dom.development.js:32373:3)
    at dispatchEvent (react-dom.development.js:30141:5)
    at dispatchDiscreteEvent (react-dom.development.js:30112:5)
onChangeAddress @ AddressCombobox.tsx:85
eval @ use-controllable.js:13
eval @ use-event.js:17
0 @ combobox.js:496
u @ match.js:11
eval @ combobox.js:494
eval @ use-event.js:17
eval @ combobox.js:579
eval @ use-event.js:17
eval @ combobox.js:659
eval @ use-event.js:17
onChange @ render.js:175
onChange @ render.js:153
callCallback @ react-dom.development.js:20565
invokeGuardedCallbackImpl @ react-dom.development.js:20614
invokeGuardedCallback @ react-dom.development.js:20689
invokeGuardedCallbackAndCatchFirstError @ react-dom.development.js:20703
executeDispatch @ react-dom.development.js:32128
processDispatchQueueItemsInOrder @ react-dom.development.js:32160
processDispatchQueue @ react-dom.development.js:32173
dispatchEventsForPlugins @ react-dom.development.js:32184
eval @ react-dom.development.js:32374
batchedUpdates$1 @ react-dom.development.js:24953
batchedUpdates @ react-dom.development.js:28844
dispatchEventForPluginEventSystem @ react-dom.development.js:32373
dispatchEvent @ react-dom.development.js:30141
dispatchDiscreteEvent @ react-dom.development.js:30112Understand this errorAI
react-dom.development.js:20724 Uncaught TypeError: Cannot read properties of null (reading 'coordinates')
    at onChangeAddress (AddressCombobox.tsx:85:25)
    at Object.eval [as current] (use-controllable.js:13:103)
    at eval (use-event.js:17:18)
    at 0 (combobox.js:496:45)
    at u (match.js:11:41)
    at Object.eval [as current] (combobox.js:494:140)
    at Object.eval [as onChange] (use-event.js:17:18)
    at Object.eval [as current] (combobox.js:579:11)
    at eval (use-event.js:17:18)
    at Object.eval [as current] (combobox.js:659:69)
    at eval (use-event.js:17:18)
    at onChange (render.js:175:42)
    at onChange (render.js:153:17)
    at HTMLUnknownElement.callCallback (react-dom.development.js:20565:14)
    at Object.invokeGuardedCallbackImpl (react-dom.development.js:20614:16)
    at invokeGuardedCallback (react-dom.development.js:20689:29)
    at invokeGuardedCallbackAndCatchFirstError (react-dom.development.js:20703:25)
    at executeDispatch (react-dom.development.js:32128:3)
    at processDispatchQueueItemsInOrder (react-dom.development.js:32160:7)
    at processDispatchQueue (react-dom.development.js:32173:5)
    at dispatchEventsForPlugins (react-dom.development.js:32184:3)
    at eval (react-dom.development.js:32374:12)
    at batchedUpdates$1 (react-dom.development.js:24953:12)
    at batchedUpdates (react-dom.development.js:28844:12)
    at dispatchEventForPluginEventSystem (react-dom.development.js:32373:3)
    at dispatchEvent (react-dom.development.js:30141:5)
    at dispatchDiscreteEvent (react-dom.development.js:30112:5)
onChangeAddress @ AddressCombobox.tsx:85
eval @ use-controllable.js:13
eval @ use-event.js:17
0 @ combobox.js:496
u @ match.js:11
eval @ combobox.js:494
eval @ use-event.js:17
eval @ combobox.js:579
eval @ use-event.js:17
eval @ combobox.js:659
eval @ use-event.js:17
onChange @ render.js:175
onChange @ render.js:153
callCallback @ react-dom.development.js:20565
invokeGuardedCallbackImpl @ react-dom.development.js:20614
invokeGuardedCallback @ react-dom.development.js:20689
invokeGuardedCallbackAndCatchFirstError @ react-dom.development.js:20703
executeDispatch @ react-dom.development.js:32128
processDispatchQueueItemsInOrder @ react-dom.development.js:32160
processDispatchQueue @ react-dom.development.js:32173
dispatchEventsForPlugins @ react-dom.development.js:32184
eval @ react-dom.development.js:32374
batchedUpdates$1 @ react-dom.development.js:24953
batchedUpdates @ react-dom.development.js:28844
dispatchEventForPluginEventSystem @ react-dom.development.js:32373
dispatchEvent @ react-dom.development.js:30141
dispatchDiscreteEvent @ react-dom.development.js:30112Understand this errorAI
```